### PR TITLE
Add test case : test case test_disk_eviction_with_node_level_soft_anti_affinity_disabled

### PR DESCRIPTION
**Pull request by cchien816**
Signed-off-by: Chris Chien <chris.chien@suse.com>

Implement test case test_disk_eviction_with_node_level_soft_anti_affinity_disabled

https://github.com/longhorn/longhorn/issues/4040

### CODE REVIEW

File: manager/integration/tests/test_node.py [hunk 1]
```diff
@@ -2611,3 +2614,80 @@ def test_disk_eviction_with_node_level_soft_anti_affinity_disabled(): # NOQA
...
+
+    # Step 6
+    replica_path = test_disk_path + '/replicas'
+    assert os.path.isdir(replica_path)
+
+    for i in range(common.RETRY_COMMAND_COUNT):
+        if len(os.listdir(replica_path)) > 0:
+            break
+        time.sleep(common.RETRY_EXEC_INTERVAL)
+    assert len(os.listdir(replica_path)) > 0
```
Comment (innobead) [inline manager/integration/tests/test_node.py:2665]: Do you think we can do this way?

- wait_for_volume_degraded
- wait_for_volume_healthy
- then just `assert len(os.listdir(replica_path)) > 0`

File: manager/integration/tests/test_node.py [hunk 2]
```diff
@@ -2611,3 +2614,80 @@ def test_disk_eviction_with_node_level_soft_anti_affinity_disabled(): # NOQA
...
+        node = client.by_id_node(lht_hostId)
+        update_disks = get_update_disks(node.disks)
+        for disk in update_disks.values():
+            if disk["path"] == test_disk_path:
+                disk.allowScheduling = False
+            else:
+                disk.allowScheduling = True
+                disk.evictionRequested = False
+
+        update_node_disks(client, node.name, disks=update_disks, retry=True)
```
Comment (innobead) [inline manager/integration/tests/test_node.py:2690]: Can we just reuse this? Probably you can move `delete_extra_disks_on_host` to common.py.

https://github.com/longhorn/longhorn-tests/blob/a7f2d31378ec3ce5aabb353f5c8618d7495b518d/manager/integration/tests/test_orphan.py#L48-L51

File: manager/integration/tests/test_node.py [hunk 3]
```diff
@@ -2611,3 +2614,80 @@ def test_disk_eviction_with_node_level_soft_anti_affinity_disabled(): # NOQA
...
+        common.cleanup_all_volumes(client)
```
Comment (innobead) [inline manager/integration/tests/test_node.py:2691]: Can we move this to the general clean up? https://github.com/longhorn/longhorn/issues/4066

**Review by innobead — CHANGES_REQUESTED** [review]

File: manager/integration/tests/test_node.py (hunk 1 above)
Comment (cchien816) [reply manager/integration/tests/test_node.py:2665]: Hi @innobead ,
From observing product behavior, when disk eviction in progress, script can not catch volume degraded status(maybe it's too quick), so currently I kept old code for review.

File: manager/integration/tests/test_node.py (hunk 2 above)
Comment (cchien816) [reply manager/integration/tests/test_node.py]: HI @innobead ,
I found we already have `cleanup_test_disks` in common can reuse.
Therefore In finalizer I only kept `cleanup_all_volumes` let no volume mounted to extra disk and let `cleanup_test_disks` can work properly at test case teardown.
Also add enable_default_disk in common teardown for review.

File: manager/integration/tests/test_node.py (hunk 3 above)
Comment (cchien816) [reply manager/integration/tests/test_node.py:2681]: HI @innobead ,
I found we already have `cleanup_test_disks` in common can reuse.
Therefore In finalizer I only kept `cleanup_all_volumes` let no volume mounted to extra disk and let `cleanup_test_disks` can work properly at test case teardown.
Also add enable_default_disk in common teardown for review.

**Review by innobead — APPROVED** [review]

**Comment by innobead** [comment]
@cchien816 Please backport to 1.3.x and 1.2.x.

**Pull request merged**